databricks/koalas/generic.py

```diff
@@ -1757,7 +1757,18 @@ def abs(self) -> Union["DataFrame", "Series"]:
         2  6  30   30
         3  7  40   50
         """
-        return self._apply_series_op(lambda kser: kser._with_new_scol(F.abs(kser.spark.column)))
+
+        def abs(kser):
+            if isinstance(kser.spark.data_type, BooleanType):
+                return kser
+            elif isinstance(kser.spark.data_type, NumericType):
+                return kser.spark.transform(F.abs)
+            else:
+                raise TypeError(
+                    "bad operand type for abs(): {}".format(kser.spark.data_type.simpleString())
```

Comment:
Hm, should we show spark's dtype instead of Spark data type?

Reply:
In that case, we should address it separately. The same error message will appear in the other places.